test/qa/test_meshnet.py

```diff
@@ -464,3 +455,64 @@ def test_invite_accept_non_existant_special_character():
         sh.nordvpn.meshnet.invite.accept("\u2222@test.com")
 
     assert "No invitation from '\u2222@test.com' was found." in str(ex.value)
+
+
+@pytest.mark.parametrize("meshnet_allias", meshnet.MESHNET_ALIAS)
+def test_set_meshnet_off_on(meshnet_allias):
+
+    assert "Meshnet is set to 'disabled' successfully." in sh.nordvpn.set(meshnet_allias, "off")
+    assert not settings.is_meshnet_on()
+
+    assert "Meshnet is set to 'enabled' successfully." in sh.nordvpn.set(meshnet_allias, "on")
+    assert settings.is_meshnet_on()
+
+
+@pytest.mark.parametrize("meshnet_allias", meshnet.MESHNET_ALIAS)
+def test_set_meshnet_on_repeated(meshnet_allias):
+
+    with pytest.raises(sh.ErrorReturnCode_1) as ex:
+            sh.nordvpn.set(meshnet_allias, "on")
+
+    assert "Meshnet is already enabled." in str(ex.value)
+
+
+@pytest.mark.parametrize("meshnet_allias", meshnet.MESHNET_ALIAS)
+def test_set_meshnet_off_repeated(meshnet_allias):
+
+    sh.nordvpn.set(meshnet_allias, "off")
+
+    with pytest.raises(sh.ErrorReturnCode_1) as ex:
+            sh.nordvpn.set(meshnet_allias, "off")
+
+    assert "Meshnet is already disabled." in str(ex.value)
+
+    sh.nordvpn.set.meshnet("on")
+
+
+@pytest.mark.parametrize("meshnet_allias", meshnet.MESHNET_ALIAS)
+def test_set_meshnet_on_logged_out(meshnet_allias):
+    
+    sh.nordvpn.logout("--persist-token")
+
+    with pytest.raises(sh.ErrorReturnCode_1) as ex:
+            sh.nordvpn.set(meshnet_allias, "on")
+
+    assert "You are not logged in." in str(ex.value)
+
+    login.login_as("default")
+    sh.nordvpn.set.meshnet("on")
+
+
+@pytest.mark.skip(reason="LVPN-4590")
+@pytest.mark.parametrize("meshnet_allias", meshnet.MESHNET_ALIAS)
+def test_set_meshnet_off_logged_out(meshnet_allias):
+    
+    sh.nordvpn.logout("--persist-token")
+
+    with pytest.raises(sh.ErrorReturnCode_1) as ex:
+            sh.nordvpn.set(meshnet_allias, "off")
+
+    assert "You are not logged in." in str(ex.value)
+
+    login.login_as("default")
```

Comment:
is `login` and `meshnet on` needed or just to check that it is successful?

Reply:
This is needed because otherwise tests would error at `teardown` because if we are logged out  and meshnet is off, we could not execute the revoking of peers and so on. Could not think of a better way to avoid this :D

Reply:
Changed to only remove peers and revoke invites at setup function instead of both setup and teardown and now those additional logins and enabling of Meshnet are not needed